### PR TITLE
do autobumps from cli

### LINKED ISSUE #63: CLI client
Create our first client to invoke Autobump. The client will be a command line interface to run Autobump.

autobump.jar --url https://bitbucket.org/grietvermeesch/testmavenproject.git

The client will:
- Git clone the project a the given url
- Find dependencies of the given project
- Look for updates of the dependencies
- Bump dependencies to the latest version by creating a branch & pull-request

### PR DESCRIPTION
**Pull request by Gschrooyen**
closes #63

### CODE REVIEW

File: autobump-core/src/test/java/com/github/autobump/core/model/usecases/AutobumpUseCaseTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    private UrlHelper urlHelper;
+
+    private URI uri;
+
+    @BeforeEach
+    void setUp() throws URISyntaxException {
+        makeMocks();
+        setUpMocks();
+    }
+    @SuppressWarnings("ExecutableStatementCount")
```
Comment (timpeeters) [inline autobump-core/src/test/java/com/github/autobump/core/model/usecases/AutobumpUseCaseTest.java:46]: Let's create a new issue on GitHub to revise this in the next sprint.
Not necessary to fix this in this pull request scope.

File: autobump-core/src/test/java/com/github/autobump/core/model/usecases/PullRequestUseCaseTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    private Bump bump;
+    private URI uri;
+
+    @BeforeEach
+    void setUp() throws URISyntaxException {
+        createMocks();
+        setUpMocks();
+    }
+
+    private void createMocks() throws URISyntaxException {
```
Comment (timpeeters) [inline autobump-core/src/test/java/com/github/autobump/core/model/usecases/PullRequestUseCaseTest.java:34]: Mockito has an Extension specific for JUnit 5 that allows to easily create mocks.

```
@ExtendWith(MockitoExtension.class)
class PullRequestUseCaseTest {
    @Mock
    private GitProvider gitProvider;

    @Mock
    private GitClient gitClient;

    @InjectInto // this creates your class under test and automatically inject mocks.
    private ClassUnderTest classUnderTest;

    // alternatively you can construct the class under test yourself in a before callback and inject manually

    void test() {
        // ...
    }
```

Not necessary to fix in this PR, but I would create a new GitHub issue to clean this up in a new sprint.

File: autobump-core/src/test/java/com/github/autobump/core/model/usecases/PullRequestUseCaseTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                .branchName(commitResult.getBranchName())
+                .title(commitResult.getCommitMessage())
+                .repoName(urlHelper.getRepoName(uri.toString()))
+                .repoOwner(urlHelper.getOwnerName(uri.toString()))
+                .build();
+        Mockito.when(gitProvider.makePullRequest(pullRequest)).thenReturn(null);
+    }
+
+    @AfterEach
+    void tearDown() {
```
Comment (timpeeters) [inline autobump-core/src/test/java/com/github/autobump/core/model/usecases/PullRequestUseCaseTest.java:59]: I would just remove empty tearDown methods().

File: build.gradle [hunk 4]
```diff
@@ -111,6 +112,7 @@ dependencies {
 }
 
 jar {
+    duplicatesStrategy = DuplicatesStrategy.EXCLUDE
```
Comment (timpeeters) [inline build.gradle:115]: Just curious, why did we run into a situation with duplicates?

Comment (Gschrooyen) [reply build.gradle:115]: this is for compliancy reasons with gradle 7.0 to suppress warnings

**Pull request merged**